stl/inc/type_traits

```diff
@@ -521,6 +488,19 @@ _INLINE_VAR constexpr bool is_volatile_v<volatile _Ty> = true;
 template <class _Ty>
 struct is_volatile : bool_constant<is_volatile_v<_Ty>> {};
 
+// STRUCT TEMPLATE is_function
+template <class _Ty>
+_INLINE_VAR constexpr bool is_function_v =
+    !is_reference_v<_Ty> && !is_const_v<const _Ty>; // determine whether _Ty is a function
+
+template <class _Ty>
+struct is_function : bool_constant<is_function_v<_Ty>> {};
+
+template <class _Ty>
+struct _Is_function { // determine whether _Ty is a function

```

Comment:
`_Weak_types` needs `_Is_function`  to provide the deprecated typedefs (`result_type`, `argument_type`, etc.); conversely nothing - after your above changes - needs `_Bool_type`. We need to restore the original implementation of `_Is_function`, and then remove the declarations of the `_Bool_type` members.